nifi-nar-bundles/nifi-standard-services/nifi-lookup-services-bundle/nifi-lookup-services/src/main/java/org/apache/nifi/lookup/DatabaseLookupService.java

```diff
@@ -0,0 +1,203 @@
+            .name("lookup-key-column")
+            .displayName("Lookup Key Column")
+            .description("Lookup key column.")
+            .required(true)
+            .addValidator(StandardValidators.NON_EMPTY_VALIDATOR)
+            .expressionLanguageSupported(true)
+            .build();
+
+    static final PropertyDescriptor LOOKUP_VALUE_COLUMN =
+        new PropertyDescriptor.Builder()
+            .name("lookup-value-column")
+            .displayName("Lookup Value Column")
+            .description("Lookup value column.")
+            .required(true)
+            .addValidator(StandardValidators.NON_EMPTY_VALIDATOR)
+            .expressionLanguageSupported(true)
+            .build();
+
+    static final PropertyDescriptor LOOKUP_NAME_COLUMN =
+        new PropertyDescriptor.Builder()
+            .name("lookup-name-column")
+            .displayName("Lookup Name Column")
+            .description("Lookup name column.")
+            .required(false)
+            .addValidator(StandardValidators.NON_EMPTY_VALIDATOR)
+            .expressionLanguageSupported(true)
+            .build();
+
+    static final PropertyDescriptor LOOKUP_NAME =
+        new PropertyDescriptor.Builder()
+            .name("lookup-name")
+            .displayName("Lookup Name")
+            .description("Lookup name.")
+            .required(false)
+            .addValidator(StandardValidators.NON_EMPTY_VALIDATOR)
+            .expressionLanguageSupported(true)
+            .build();
+
+    private List<PropertyDescriptor> properties;
+
+    private volatile BasicConfigurationBuilder<DatabaseConfiguration> builder;
+
+    private Configuration getConfiguration() {
+        try {
+            if (builder != null) {
+                return builder.getConfiguration();
+            }
+        } catch (final ConfigurationException e) {
+            getLogger().error(e.getMessage(), e);
+        }
+        return null;
+    }
+
+    @Override
+    protected List<PropertyDescriptor> getSupportedPropertyDescriptors() {
+        return properties;
+    }
+
+    @Override
+    protected void init(final ControllerServiceInitializationContext context) throws InitializationException {
+        super.init(context);
+        final List<PropertyDescriptor> properties = new ArrayList<>(super.getSupportedPropertyDescriptors());
+        properties.add(CONNECTION_POOL);
+        properties.add(LOOKUP_TABLE_NAME);
+        properties.add(LOOKUP_KEY_COLUMN);
+        properties.add(LOOKUP_VALUE_COLUMN);
+        properties.add(LOOKUP_NAME_COLUMN);
+        properties.add(LOOKUP_NAME);
+        this.properties = Collections.unmodifiableList(properties);
+    }
+
+    @OnEnabled
+    public void onEnabled(final ConfigurationContext context) throws InitializationException {
+        final DBCPService databaseService = context.getProperty(CONNECTION_POOL).asControllerService(DBCPService.class);
+        final DataSource dataSource = databaseService.getDataSource();
```

Comment:
As discussed at #1450, I think we'd better to avoid exposing DataSource from DBCPService. Wrapping Connection to be a DataSource can be done here without requiring DBCPService change. Thought?